src/nsmgr/internal/pkg/manager/manager.go

```diff
@@ -0,0 +1,151 @@
+package manager
+
+import (
+	"context"
+	"net"
+	"os"
+	"os/signal"
+	"path"
+	"syscall"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/networkservicemesh/cmd-nsmgr/src/nsmgr/internal/pkg/constants"
+	"github.com/networkservicemesh/cmd-nsmgr/src/nsmgr/internal/pkg/deviceplugin"
+	"github.com/networkservicemesh/cmd-nsmgr/src/nsmgr/internal/pkg/flags"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/chains/nsmgr"
+	"github.com/networkservicemesh/sdk/pkg/tools/grpcoptions"
+	"github.com/networkservicemesh/sdk/pkg/tools/spanhelper"
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/go-spiffe/spiffe"
+	"google.golang.org/grpc"
+)
+
+func RunNsmgr(ctx context.Context, values *flags.DefinedFlags) error {
+	span := spanhelper.FromContext(ctx, "run")
+	defer span.Finish()
+	ctx = span.Context()
+
+	var err error
+	// Capture signals to cleanup before exiting - note: this *must* be the first thing in main
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs,
+		os.Interrupt,
+		// More Linux signals here
+		syscall.SIGHUP,
+		syscall.SIGTERM,
+		syscall.SIGQUIT)
+
+	// Setup logging
+	logrus.SetReportCaller(true)
+
+	// Context to use for all things started in main
+	var cancelFunc context.CancelFunc
+	ctx, cancelFunc = context.WithCancel(ctx)
+	defer cancelFunc()
+
+	var spiffieTlsPeer *spiffe.TLSPeer
+	span.LogValue("Insecure", values.Insecure)
+	if !values.Insecure {
+		spiffieTlsPeer, err = spiffe.NewTLSPeer(spiffe.WithWorkloadAPIAddr(values.SpiffeAgentURL.String()))
```

Comment:
Please use spiffeutils.NewTLSPeer and deprecate the insecure flag.

Reply:
@edwarnicke  I can't make it working without SpiffieAgentURL.